### PR TITLE
Max Chapter3/homework/5

### PR DESCRIPTION
**Pull request by HellHunterMax**

### CODE REVIEW

File: Src/BootCamp.Chapter/MostCommonLetterFinder.cs [hunk 1]
```diff
@@ -1,10 +1,89 @@
...
+namespace BootCamp.Chapter
 {
     public static class MostCommonLetterFinder
     {
         public static char Find(string sentence)
         {
-            return 'z';
+            if (string.IsNullOrWhiteSpace(sentence))
+            {
+                throw new ArgumentNullException();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/MostCommonLetterFinder.cs:12]: missing giving some context- what argument was null?

File: Src/BootCamp.Chapter/MostCommonLetterFinder.cs [hunk 2]
```diff
@@ -1,10 +1,89 @@
...
+            }
+
+            List<char> sentenceList = new List<char>();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/MostCommonLetterFinder.cs:15]: This list is not needed- a string can be treated as array of char, except that you cannot modify individual character, however you can read individual characters and certainly foreach it.

File: Src/BootCamp.Chapter/MostCommonLetterFinder.cs [hunk 3]
```diff
@@ -1,10 +1,89 @@
...
+        }
+
+        private static char GetMostCommonLetter(string originalSentence, List<char> sortedSentenceList)
+        {
+            int mostCommonLetterCount = 0;
+            char mostCommonLetter = ' ';
+            int nextLetterCount = 0;
+            char nextLetter = ' ';
+
+            foreach (char c in sortedSentenceList)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/MostCommonLetterFinder.cs:32]: Interesting solution. However, you can make it much more simple. Have you considered using a dictionary? :)

File: Src/BootCamp.Chapter/ToggleableGrid2d.cs [hunk 4]
```diff
@@ -1,16 +1,58 @@
...
     public class ToggleableGrid2D : IToggleableGrid
     {
         private readonly IGridClearer _gridClearer;
+        bool[,] grid;
 
         public ToggleableGrid2D(bool[,] toggles, IGridClearer gridClearer)
         {
             _gridClearer = gridClearer;
+            grid = toggles;
+
```
Comment (Almantask) [inline Src/BootCamp.Chapter/ToggleableGrid2d.cs:14]: No need a blank

File: Src/BootCamp.Chapter/ToggleableGrid2d.cs [hunk 5]
```diff
@@ -1,16 +1,58 @@
...
+                    }
+                }
+                if (i != grid.GetLength(0) -1)
+                {
+                    Console.WriteLine();
+                }
+            }
+        }
+
+        private void OutOfBoundsCheck(int x, int y)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/ToggleableGrid2d.cs:46]: ``CheckIsOutOfBounds`` is a better name

File: Src/BootCamp.Chapter/MostCommonLetterFinder.cs (hunk 2 above)
Comment (HellHunterMax) [reply Src/BootCamp.Chapter/MostCommonLetterFinder.cs:15]: yes i knew that, I did it so I could sort the list. but removed it in the new version with dictionary.

File: Src/BootCamp.Chapter/MostCommonLetterFinder.cs [hunk 6]
```diff
@@ -9,81 +9,40 @@ public static char Find(string sentence)
         {
             if (string.IsNullOrWhiteSpace(sentence))
             {
-                throw new ArgumentNullException();
+                throw new ArgumentNullException("The sentence you gave was empty.");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/MostCommonLetterFinder.cs:12]: This is fine, homeork normally you just pass a name of parameter using ``nameof(sentence)``

File: Tests/BootCamp.Chapter.Tests/MostCommonLetterFinderTests.cs [hunk 7]
```diff
@@ -23,6 +23,7 @@ public void Find_Given_NullOrWhitespace_Throws_ArgumentNullExceptions(string sen
         [InlineData("abca", 'a')]
         [InlineData("uiopppaaaxxx", 'p')]
         [InlineData("II", 'I')]
+        [InlineData("sfajlasbfklasjksahfksjbjbfkjahbgkfiyugtzvduytwerfgearkjbmnzvjhgliuwyeiuflkjfsjhvbmnxbvjugswugakjhsfksfjhkjbafkujhb", 'j')]
```
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/MostCommonLetterFinderTests.cs:26]: Awesome, getting used to writing your own test cases :)

**Pull request closed without merging**